spec/rspec/mocks/reraising_eager_raises_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "spec_helper"
+
+RSpec.describe "Reraising eager raises during the verify step" do
+  it "does not reraise when a double receives a message that hasn't been allowed/expected" do
+    expect {
+      with_unfulfilled_double do |dbl|
+        begin
+          dbl.foo
+        rescue RSpec::Mocks::MockExpectationError
+          RSpec::Mocks.verify
+        end
+      end
+    }.not_to raise_error
```

Comment:
Putting the `verify` in the `rescue` clause is a little weird -- consider that if `RSpec::Mocks::MockExpectationError` is not raised, the verify will never happen, and this spec will pass even though it's not doing what you expect.  It would be a false positive.  Instead, can you use this pattern?

``` ruby
expect { dbl.foo }.to fail
expect { verify_all }.not_to raise_error
```

It's shorter too :).  `fail` is defined in `spec/support/matchers.rb`.  `verify_all` is defined in `spec_helper`.